src/main/java/dev/morling/onebrc/CalculateAverage_abhinavupadhyay.java

```diff
@@ -0,0 +1,256 @@
+/*
+ *  Copyright 2023 The original authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package dev.morling.onebrc;
+
+import sun.misc.Unsafe;
+
+import java.io.IOException;
+import java.lang.foreign.Arena;
+import java.lang.reflect.Field;
+import java.nio.channels.FileChannel;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class CalculateAverage_abhinavupadhyay {
+
+    private static final String FILE_NAME = "./measurements.txt";
+
+    private static Unsafe initUnsafe() {
+        try {
+            final Field theUnsafe = Unsafe.class.getDeclaredField("theUnsafe");
+            theUnsafe.setAccessible(true);
+            return (Unsafe) theUnsafe.get(Unsafe.class);
+        }
+        catch (NoSuchFieldException | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private final static Unsafe UNSAFE = initUnsafe();
+
+    public static final long HASSEMI = 0x3B3B3B3B3B3B3B3BL;
+
+    static long has0(long x) {
+        return (x - 0x0101010101010101L) & (~x) & 0x8080808080808080L;
+    }
+
+    private static class Table {
+        private static final int TABLE_SIZE = 1 << 21; // 0x8000; // collisions with table smaller than this. Need a better hash function
+        private static final int TABLE_MASK = TABLE_SIZE - 1;
+        Row[] table = new Row[TABLE_SIZE];
+        byte[] array = new byte[256];
+
+        public void put(long cityStartOffset, long cityLength, long nameHash, int temperature) {
+            final long uhash = nameHash ^ (nameHash >> 29);
+            // final long uhash = nameHash;
+            int index = (int) uhash & TABLE_MASK;
+            Row row = table[index];
```

Comment:
You'll need to handle collisions, otherwise you'd aggregate the values from several stations with the same hash.